libs/mve/image_io.cc

```diff
@@ -628,6 +635,104 @@ tiff_error_handler (char const* /*module*/, char const* fmt, va_list ap)
     throw util::Exception(msg);
 }
 
+ImageHeaders
+load_tiff_file_headers (std::string const& filename)
+{
+    ImageHeaders headers;
+    TIFFSetWarningHandler(nullptr);
+    TIFFSetErrorHandler(tiff_error_handler);
+
+    TIFF* tif = TIFFOpen(filename.c_str(), "r");
+    if (!tif)
+        throw util::Exception("TIFF file format not recognized");
```

Comment:
Would that be a `util::FileException`?

Reply:
Good catch, yes it would be better instead of the generic one. :+1: